src/gui/gui_topbar.hpp

```diff
@@ -1518,27 +1518,27 @@ class topbar_colony_icon : public standard_nation_button {
 	}
 
 	void update_tooltip(sys::state& state, int32_t x, int32_t y, text::columnar_layout& contents) noexcept override {
-		if(parent) {
-			Cyto::Any payload = dcon::nation_id{};
-			parent->impl_get(state, payload);
-			auto nation_id = any_cast<dcon::nation_id>(payload);
-
-			auto nation_fat_id = dcon::fatten(state.world, nation_id);
-			
-			if(nations::can_expand_colony(state, nation_id)) {
-				nation_fat_id.for_each_colonization([&](dcon::colonization_id colony) {
-					auto colState = dcon::fatten(state.world, colony).get_state();
-					text::add_line(state, contents, "countryalert_colonialgood_state", text::variable_type::region, colState);
-				});
-			} else if(nations::is_losing_colonial_race(state, nation_id)) {
-				nation_fat_id.for_each_colonization([&](dcon::colonization_id colony) {
-					auto colState = dcon::fatten(state.world, colony).get_state();
-					text::add_line(state, contents, "countryalert_colonialbad_influence", text::variable_type::region, colState);
-				});
-			} else {
-				text::add_line(state, contents, "countryalert_no_colonial");
+		auto nation_id = retrieve<dcon::nation_id>(state, parent);
+		auto nation_fat_id = dcon::fatten(state.world, nation_id);
+
+		bool is_empty = true;
+		nation_fat_id.for_each_colonization([&](dcon::colonization_id colony) {
+			auto sdef = state.world.colonization_get_state(colony);
+			if(province::can_invest_in_colony(state, nation_id, sdef)) {
```

Comment:
you also need a check for when the nation can end the colonial race and turn it into a colony. This is not covered by can invest (check get_colonization_stage == 3)